### PR TITLE
feat: load cert and key for TLS via `@link`

### LINKED ISSUE #1121: Use link to load cert and key
Currently cert and key are given as file paths and are part of `Server`. The proposal is to move this implementation from `@server` to into the newly launched `@link` operator.

```graphql
schema
  @link(type: Cert src: "abc.cert")
  @link(type: Key src: "key.cert") {
  query: Query
}
```

### PR DESCRIPTION
**Pull request by ologbonowiwi**
- chore: delete `dbg!` log
- feat: drop cert and key fields in favour of `Link` usage
- style: run `./lint.sh --mode=fix`

**Issue Reference(s):**  
Fixes #1121
/claim #1121

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/config/reader.rs [hunk 1]
```diff
@@ -135,12 +140,52 @@ impl ConfigReader {
...
+                        Arc::new(self.load_private_key(content.clone()).await?)
+                }
             }
         }
 
         Ok(config_set)
     }
 
+    /// Reads the certificate from a given file
+    pub async fn load_cert(&self, content: String) -> anyhow::Result<Vec<CertificateDer<'static>>> {
```
Comment (tusharmath) [inline src/config/reader.rs:157]: does not need to be public.

File: src/config/reader.rs [hunk 2]
```diff
@@ -135,12 +140,52 @@ impl ConfigReader {
...
+        let certificates = rustls_pemfile::certs(&mut content.as_bytes())?;
+
+        Ok(certificates.into_iter().map(CertificateDer::from).collect())
+    }
+
+    /// Reads a private key from a given file
+    async fn load_private_key(
```
Comment (tusharmath) [inline src/config/reader.rs:164]: does not need to be public.

File: src/blueprint/server.rs [hunk 3]
```diff
@@ -40,7 +42,10 @@ pub struct Script {
 #[derive(Clone, Debug)]
 pub enum Http {
     HTTP1,
-    HTTP2 { cert: String, key: String },
+    HTTP2 {
+        cert: Vec<CertificateDer<'static>>,
```
Comment (tusharmath) [inline src/blueprint/server.rs:46]: Why do we need a Vec?

File: src/blueprint/links.rs [hunk 4]
```diff
@@ -39,6 +39,18 @@ impl TryFrom<Vec<Link>> for Links {
...
+                .iter()
+                .filter(|l| l.type_of == LinkType::Key)
+                .collect::<Vec<&Link>>();
+
+            if key_links.len() > 1 {
+                Valid::fail("Only one key link is allowed".to_string())
+            } else {
+                Valid::succeed(links)
+            }
+        })
```
Comment (tusharmath) [inline src/blueprint/links.rs:53]: also only one cert is allowed?

Comment (ologbonowiwi) [reply src/blueprint/links.rs:53]: More than one is accepted.

File: src/blueprint/server.rs (hunk 3 above)
Comment (ologbonowiwi) [reply src/blueprint/server.rs:46]: `rustls_pemfile::read` returns a vec and the tls acceptor also expects a vec

File: src/blueprint/server.rs [hunk 5]
```diff
@@ -40,7 +42,10 @@ pub struct Script {
...
+        key: Arc<PrivateKeyDer<'static>>,
```
[the remark this answers is not in the document]
Comment (ologbonowiwi) [reply src/blueprint/server.rs:47]: because `PrivateKeyDer` does not derive from `Clone`

**Review by tusharmath — APPROVED** [review]

**Pull request merged**